urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/PairsInAv231.java

```diff
@@ -0,0 +1,243 @@
+        doWilf(3,4, true, 10);
+        // doSpecial();
+    }
+
+    public static void doSpecial() {
+
+        PermutationClass c = new PermutationClass("231", "321");
+        long[] cCounts = new long[8];
+        for (int n = 4; n < 12; n++) {
+            PermCounter counter = new PermCounter();
+            c.processPerms(n, counter);
+            cCounts[n - 4] = counter.getCount();
+        }
+        System.out.println(Arrays.toString(cCounts));
+        c = new PermutationClass("231", "123");
+        for (int n = 4; n < 12; n++) {
+            PermCounter counter = new PermCounter();
+            c.processPerms(n, counter);
+            cCounts[n - 4] = counter.getCount();
+        }
+        System.out.println(Arrays.toString(cCounts));
+        c = new PermutationClass("231", "321");
+        for (Permutation q : new Permutations(c, 4)) {
+            PermutationClass d = new PermutationClass("231", "321", q.toString());
+            for (int n = 4; n < 12; n++) {
+                PermCounter counter = new PermCounter();
+                d.processPerms(n, counter);
+                cCounts[n - 4] = counter.getCount();
+            }
+            System.out.println(q + " " + Arrays.toString(cCounts));
+        }
+    }
+
+    public static void doWilf(int n, boolean verbose, int checkTotal) {
+        PermutationClass c = new PermutationClass("231");
+        HashSet<PermPair> pairs = new HashSet<>();
+        for (Permutation a : new Permutations(c, n)) {
+            for (Permutation b : new Permutations(c, n)) {
+                if (!a.equals(b)) {
+                    pairs.add(new PermPair(a, b));
+                }
+            }
+        }
+        HashSet<PermPair> pairReps = new HashSet<>();
+        for (PermPair pair : pairs) {
+            PermPair np = new PermPair(Symmetry.IRC.on(pair.a), Symmetry.IRC.on(pair.b));
+            if (!pairReps.contains(np) && !pairReps.contains(pair)) {
+                pairReps.add(pair);
+            }
+        }
+        System.out.println("Length " + n + " has " + pairReps.size() + " symmetry orbits.");
+        HashMap<Spectrum, HashSet<PermPair>> wilfClasses = new HashMap<>();
+        for (PermPair pair : pairs) {
+            long[] qC = new long[checkTotal];
+            PermutationClass d = new PermutationClass(new Permutation("231"), pair.a, pair.b);
+            for (int m = n + 1; m <= n + checkTotal; m++) {
+                PermCounter counter = new PermCounter();
+                d.processPerms(m, counter);
+                qC[m - n - 1] = counter.getCount();
+            }
+            // System.out.println(q + " " + Arrays.toString(qC));
+            Spectrum qs = new Spectrum(qC);
+            if (wilfClasses.containsKey(qs)) {
+                wilfClasses.get(qs).add(pair);
+            } else {
+                HashSet<PermPair> aSet = new HashSet<>();
+                aSet.add(pair);
+                wilfClasses.put(qs, aSet);
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 85:91, 138:144.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.